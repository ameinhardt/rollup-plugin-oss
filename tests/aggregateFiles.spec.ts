/* eslint-disable eslint-comments/disable-enable-pair */
import { createRequire } from 'node:module';
import { dirname } from 'node:path';
import { fileURLToPath } from 'node:url';
import { aggregateFiles } from '../index';

const { resolve } = createRequire(dirname(fileURLToPath(import.meta.url)));

describe('gets text for license definition', function() {
  it('fixed given license text', async function() {
    const libraries = await aggregateFiles(
      ['tslib', '@ameinhardt/eslint-config', '@ameinhardt/eslint-config/typescript'].map((file) => resolve(file))
    );
    expect(libraries).toBeInstanceOf(Array);
    for (const value of libraries) {
      expect(value).toEqual(expect.objectContaining({
        author: expect.any(String),
        license: expect.any(String),
        licenseText: expect.any(String),
        name: expect.any(String),
        repository: expect.any(String),
        version: expect.any(String)
      }));
    }
  });

  it('ignore unknown package', async function() {
    const libraries = await aggregateFiles(['@one/too/three']);
    expect(libraries.length).toBe(0);
  });

  it('ignore filtered package', async function() {
    const libraries = await aggregateFiles([resolve('tslib')], undefined, /ts/);
    expect(libraries.length).toBe(0);
  });

  it('ignore filtered package', async function() {
    const packerArguments : Array<string> = [],
      result = await aggregateFiles([resolve('tslib')], (data, filename) => packerArguments.push(filename));
    expect(result).toMatchSnapshot();
    expect(packerArguments).toMatchSnapshot();
  });
});
