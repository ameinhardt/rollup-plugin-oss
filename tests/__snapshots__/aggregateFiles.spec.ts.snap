// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets text for license definition ignore filtered package 1`] = `
[
  {
    "author": "Microsoft Corp.",
    "description": "Runtime library for TypeScript helper functions",
    "license": "0BSD",
    "licenseText": "Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.",
    "name": "tslib",
    "repository": "https://github.com/Microsoft/tslib",
    "version": "2.5.0",
  },
]
`;

exports[`gets text for license definition ignore filtered package 2`] = `
[
  "tslib@2.5.0/package.json",
  "tslib@2.5.0/LICENSE.txt",
  "tslib@2.5.0/tslib.js",
]
`;
